GNOME #436797; Don't start a timed login if an username was already entered.

Index: gdm-2.18.2/daemon/verify-pam.c
===================================================================
--- gdm-2.18.2.orig/daemon/verify-pam.c	2007-05-29 10:38:01.000000000 +0200
+++ gdm-2.18.2/daemon/verify-pam.c	2007-05-29 10:38:02.000000000 +0200
@@ -830,17 +830,18 @@
     started_timer = FALSE;
     null_tok = 0;
 
-    /* start the timer for timed logins */
-    if ( ! ve_string_empty (gdm_get_value_string (GDM_KEY_TIMED_LOGIN)) &&
-	d->timed_login_ok &&
-	(local || gdm_get_value_bool (GDM_KEY_ALLOW_REMOTE_AUTOLOGIN))) {
-	    gdm_slave_greeter_ctl_no_ret (GDM_STARTTIMER, "");
-	    started_timer = TRUE;
-    }
-
+    /* don't start a time login if we've already entered a username */
     if (username != NULL) {
 	    login = g_strdup (username);
 	    gdm_slave_greeter_ctl_no_ret (GDM_SETLOGIN, login);
+    } else {
+	    /* start the timer for timed logins */
+	    if ( ! ve_string_empty (gdm_get_value_string (GDM_KEY_TIMED_LOGIN)) &&
+		d->timed_login_ok &&
+		(local || gdm_get_value_bool (GDM_KEY_ALLOW_REMOTE_AUTOLOGIN))) {
+		    gdm_slave_greeter_ctl_no_ret (GDM_STARTTIMER, "");
+		    started_timer = TRUE;
+		}
     }
 
     cur_gdm_disp = d;
